File: 2024/2A.cpp
#include <bits/stdc++.h>
#include "readfile.hpp"


int main(int argc, char const *argv[])
{
	int res = 0;
	std::vector<std::string> file = readfile("2A.in");
	if (file.empty())
		return 1;

	for (const auto &line : file) {
		std::istringstream iss(line);
		int n;
		bool safe = true;
		int prev = -1;
		bool increasing = false, decreasing = false;
		// put to vector and pair of two
		while (iss >> n) {
			if (prev == -1) {
				prev = n;
				continue;
			}
			if (prev < n)
				increasing = true;
			else if (prev > n)
				decreasing = true;
			if (std::abs(prev - n) > 3 || prev == n || (n > prev && decreasing || n < prev && increasing || increasing && decreasing || !increasing && !decreasing)) {
				safe = false;
				break;
			}
			prev = n;
		}
		if (safe) {
			res++;
		}
	}

	// print
	std::cout << res << std::endl;
}

File: 2024/2A.py
res = 0

with open("2.in") as f:
    for i in f:
        increasing = False
        decreasing = False
        safe = True
        n = i.split()
        prev = -1
        for j in n:
            j = int(j)
            if prev == -1:
                prev = j
                continue
            if prev < j:
                increasing = True
            elif prev > j:
                decreasing = True
            if (abs(prev - j) > 3 or prev == j or (j > prev and decreasing or j < prev and increasing or increasing and decreasing or not increasing and not decreasing)):
                safe = False
                # break
            prev = j
        if safe:
            res += 1

print(res)
